### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#9]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 Alright, understood. Based on your description, it seems you are looking for a way to use the `wandb.run.name` into the checkpoint filename with more structured format like `swept_sky_54-epoch10-loss0.54`. In the current framework, you are right, as ModelCheckpoint takes only optimizer metrics (like `val_loss`) by default.
However, you can achieve this by overriding the `format_checkpoint_name` function in ModelCheckpoint callback. Here's below how to do it:
```python
import pytorch_lightning as pl
class CustomModelCheckpoint(pl.callbacks.ModelCheckpoint):
    def format_checkpoint_name(self, version, metrics, ver_overflow):
        # If using wandb, pass wandb run name
        if 'wandb' in globals():
            wandb_run_name = wandb.run.name
            filename = f"{wandb_run_name}-{version}-loss{metrics['val_loss']:.2f}"
            return filename
        else:
            # If not using wandb
            return super().format_checkpoint_name(version, metrics, ver_overflow)
# example use
checkpoint_callback = CustomModelCheckpoint(dirpath="my/path/")
trainer = pl.Trainer(callbacks=[checkpoint_callback])
```
This should give you the filename with the wandb run name included as you want. Please, note that this version of code will only work if `wandb` (Weights & Biases) is being used. So make sure to use `wandb` along with this.

<details open>
<summary>Checklist</summary>

- [X] Modify `src/lightning/pytorch/callbacks/model_checkpoint.py` ✓ https://github.com/ayulockin/pytorch-lightning/commit/b9622a784a7533be1e8b365753e9700ebcb5e46b [Edit](https://github.com/ayulockin/pytorch-lightning/edit/sweep/sweep_format_wandb_run_name_in_pytorch/src/lightning/pytorch/callbacks/model_checkpoint.py#L567-L605)
- [X] Running GitHub Actions for `src/lightning/pytorch/callbacks/model_checkpoint.py` ✓  [Edit](https://github.com/ayulockin/pytorch-lightning/edit/sweep/sweep_format_wandb_run_name_in_pytorch/src/lightning/pytorch/callbacks/model_checkpoint.py#L567-L605)
</details>
